### PR TITLE
Re: issues 1356. Eliminated reexport of serde_json from a number of m…

### PR DESCRIPTION
**Pull request by thessc1**
…odules. Also,reexported a number of 3rd party dependencies for fedimint-api.

How does this work?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ it
 .tmpenv
 .direnv
 .vscode
+.DS_Store
```
Comment (elsirion) [inline .gitignore:11]: Please do not make unrelated edits in PRs. You can ignore paths locally by adding them to `.git/info/exclude`. If you think this should be added to the global `.gitignore` please open a separate PR (although I feel more like removing idea and vscode looking at it, these are just dev-specific and should not be global).

**Review by elsirion — COMMENTED** [review]
Otherwise looks good :)

File: .gitignore (hunk 1 above)
Comment (thessc1) [reply .gitignore]: Roger. Removed.

File: fedimint-server/Cargo.toml [hunk 2]
```diff
@@ -20,12 +20,14 @@ bitcoin = "0.29.2"
...
+fedimint-api = { path = "../fedimint-api" }
+fedimint-core = { path = "../fedimint-core" }
+fedimint-wallet = { path = "../modules/fedimint-wallet", features = ["native"] }
 futures = "0.3.24"
 impl-tools = "0.6.1"
 itertools = "0.10.5"
 jsonrpsee = { version = "0.16.2", features = ["server"] }
-fedimint-api = { path = "../fedimint-api" }
-fedimint-core = { path = "../fedimint-core" }
+mint-client = { path = "../client/client-lib" }
```
Comment (elsirion) [inline fedimint-server/Cargo.toml:30]: Why does the server rely on the mint client? Maybe that's a bit too involved and should go into a separate PR, but would be good to know. (pinging @jkitman who also worked on modularization)

Comment (jkitman) [reply fedimint-server/Cargo.toml:30]: We just finished a refactor to avoid the `fedimint-server` depending on modules, we should keep it that way.

**Comment by thessc1** [comment]
Hey folks, anything else needed here? Happy to resolve the conflicts if needed.

**Review by dpc — APPROVED** [review]

**Review by jkitman — CHANGES_REQUESTED** [review]
Beyond the conflicts we cannot have `fedimint-server` depend on `mint-client` again.

**Comment by dpc** [comment]
>
